### PR TITLE
add unary operators 'lowercase' and 'uppercase'

### LINKED ISSUE #540: [FEATURE] lowercase/uppercase  operator
## User Story
1. As a {user}
2. I want {to use a lowercase/uppercase operator}
3. So that {I can consistently make a string lowercase/uppercase}

## User Acceptance Criteria
- [ ] A `lowercase`/`uppercase` operator is available, using the `toLowerCase`/`toUpperCase` method from Typescript

## Examples
```
transform LowerCaseText {
    from inputValue oftype text;
    to outputValue oftype text;

    outputValue: lowercase inputValue;
}
```

## Notes

- Operator example PRs
  - https://github.com/jvalue/jayvee/pull/521
  - https://github.com/jvalue/jayvee/pull/526
- How to test
  - https://github.com/jvalue/jayvee/blob/main/package.json#L12
  - https://github.com/jvalue/jayvee/blob/main/.vscode/launch.json#L9

## Definitions of Done
- [ ] A PR has been opened and accepted
- [ ] All user acceptance criteria are met
- [ ] All tests are passing

### PR DESCRIPTION
**Pull request by TungstnBallon**
adds the `lowercase` and  `lowercase` operators.

I adapted `cars.jv* to demonstrate them. This file adds 2 new columns, one for the lowercase and one for the uppercase name.

```
pipeline CarsPipeline {

	CarsExtractor
        -> CarsTextFileInterpreter
		-> CarsCSVInterpreter
		-> NameHeaderWriter
	   	-> CarsTableInterpreter
        -> NameToLowerCase
        -> NameToUpperCase
		-> CarsLoader;

    block CarsExtractor oftype HttpExtractor {
	url: "https://gist.githubusercontent.com/noamross/e5d3e859aa0c794be10b/raw/b999fb4425b54c63cab088c0ce2c0d6ce961a563/cars.csv";
	}

	block CarsTextFileInterpreter oftype TextFileInterpreter { }

	block CarsCSVInterpreter oftype CSVInterpreter {
		enclosing: '"';
	}

	block NameHeaderWriter oftype CellWriter {
		at: cell A1;
		write: ["name"];
	}

	block CarsTableInterpreter oftype TableInterpreter {
		header: true;
		columns: [
			"name" oftype text,
			"mpg" oftype decimal,
			"cyl" oftype integer,
			"disp" oftype decimal,
			"hp" oftype integer,
			"drat" oftype decimal,
			"wt" oftype decimal,
			"qsec" oftype decimal,
			"vs" oftype integer,
			"am" oftype integer,
			"gear" oftype integer,
			"carb" oftype integer
		];
	}


	transform LowerCaseText {
		from inputValue oftype text;
		to outputValue oftype text;

		outputValue: lowercase inputValue;
	}

    transform UpperCaseText {
		from inputValue oftype text;
		to outputValue oftype text;

		outputValue: uppercase inputValue;
	}

    block NameToLowerCase oftype TableTransformer {
        inputColumns: ["name"];
        outputColumn: "name_lowercase";

        use: LowerCaseText;
    }

    block NameToUpperCase oftype TableTransformer {
        inputColumns: ["name"];
        outputColumn: "NAME_UPPERCASE";

        use: UpperCaseText;
    }

	block CarsLoader oftype SQLiteLoader {
		table: "Cars";
		file: "./cars.sqlite";
	}
}
```

closes #540 
@rhazn

### CODE REVIEW

**Comment by rhazn** [comment]
Great, thanks! Please first sign the CLA (commenting the sentence the bot asks you to, then retriggering the check). Then you can add me as a reviewer (top right) and I'll get a notification to review this asap :).

**Comment by rhazn** [comment]
Looks good, now last thing left is officially requesting a review from me (top right, reviewers) and then I'll try to get to it asap.

If you have time left in the meantime, you can familiarize yourself with the type system a bit by trying to understand if we can allow string and regex values for a parameter (for this task https://github.com/jvalue/jayvee/issues/523).

**Comment by georg-schwarz** [comment]
Hey @TungstnBallon, I merged some changes in #539 that affect your changes. I suggest doing a `git rebase` to incorporate the changes.
I think you only need to update the import paths and imported classes/variables in `string-transform-type-computer.ts` to adapt your feature. 

Let me know if you need help with that ;-)

File: apps/docs/docs/user/expressions.md
Comment (rhazn) [inline apps/docs/docs/user/expressions.md:63]: I try to talk about `text`, not `string` because our value type for strings is called text.

```suggestion
- `lowercase` converts all alphabetic characters in a text to lowercase
- `uppercase` converts all alphabetic characters in a text to uppercase
```

**Review by rhazn — APPROVED** [review]
Looks good to me 👍 .

Regarding review flow, what I do is:
- Approve with comments: You can merge at any time, fixing the comments would be good but I'll leave it up to your best judgement how / if it makes sense.
- Request changes: Don't merge before the comments are resolved. Once you consider the comments resolved, re-request a new review from the PR main page.

So in this case, you can use your best judgement if you want to accept my suggested changes or not and afterwards you are free to merge without a new review.

**Pull request merged**
